### PR TITLE
[bitnami/rails] Add Rails container VIB tests

### PR DESCRIPTION
**Pull request by mdhont**
### Description of the change

Adding the VIB tests for Rails container

### CODE REVIEW

File: .vib/rails/goss/rails.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+command:
```
Comment (FraPazGal) [inline .vib/rails/goss/rails.yaml:1]: I would add a check for the database-related gems added by us at compilation time.

File: .vib/rails/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+binaries:
```
Comment (FraPazGal) [inline .vib/rails/goss/vars.yaml:1]: You are missing a lot of subcomponents added in the Dockerfile. Please check the [subcomponent list](https://github.com/bitnami/containers/blob/main/bitnami/rails/7/debian-11/prebuildfs/opt/bitnami/.bitnami_components.json).

File: .vib/rails/goss/rails.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  # A new Rails project can be created
+  check-project:
```
Comment (FraPazGal) [inline .vib/rails/goss/rails.yaml:3]: Nice test, though I would include a server initialization here.

File: .vib/rails/goss/vars.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  - rails
+  - ruby
+directories:
+  - mode: "0755"
+    paths:
+      - /opt/bitnami/ruby/lib
+      - /opt/bitnami/ruby/bin
```
Comment (FraPazGal) [inline .vib/rails/goss/vars.yaml:8]: I don't think we are modifying either of these folders' permissions. We are only doing it so for [/app](https://github.com/bitnami/containers/blob/main/bitnami/rails/7/debian-11/rootfs/opt/bitnami/scripts/rails/postunpack.sh), which should be included here.

**Review by FraPazGal — CHANGES_REQUESTED** [review]
I added some comments here @mdhont, please check them when you can.

File: .vib/rails/goss/rails.yaml (hunk 3 above)
Comment (mdhont) [reply .vib/rails/goss/rails.yaml:3]: I've added server initialization, it needs to be in the same test, as Goss tests aren't executed in the listed order.

File: .vib/rails/goss/vars.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+binaries:
+  - gosu
+  - mysql
+  - node
+  - python
+  - rails
+  - ruby
+  - wait-for-port
```
Comment (FraPazGal) [inline .vib/rails/goss/vars.yaml:8]: We usually do not add `wait-for-port` here, as we understand that it is only a minor tool used at initialization time. As such, the existence of this binary is implicitly tested when the container is fully deployed. No strong thought on this though, wdyt @joancafom?

File: .vib/rails/goss/rails.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    timeout: 120000
+    exec: "rails new /app && cd /app && timeout 10s rails s -p 4000 || true"
+    exit-status: 0
+    stdout:
+      - Listening
+  check-app-version:
+  # rails --version returns other version format than APP_VERSION
```
Comment (FraPazGal) [inline .vib/rails/goss/rails.yaml:10]: It'll be great if you could expand this comment a bit and give an overview of the syntax used by the asset here.

**Review by FraPazGal — COMMENTED** [review]
Added a couple more comments!

**Review by FraPazGal — APPROVED** [review]
LGTM!

**Pull request merged**